crates/forge/src/shared_cache.rs

```diff
@@ -0,0 +1,53 @@
+pub const CACHE_DIR: &str = ".snfoundry_cache";
+
+use anyhow::Result;
+use camino::Utf8PathBuf;
+use scarb_metadata::MetadataCommand;
+use std::fs::{self, File};
+use std::io::{BufRead, BufReader, Write};
+
+use crate::test_case_summary::TestCaseSummary;
+
+pub fn read_tests_failed_file() -> Result<Vec<String>> {
+    let tests_failed_path = cache_dir()?.join(".prev_tests_failed");
+    let file = File::open(tests_failed_path)?;
+    let buf = BufReader::new(file);
+    Ok(buf
+        .lines()
+        .map(|l| l.expect("Could not parse line"))
+        .collect())
+}
+
+pub fn cache_dir() -> Result<Utf8PathBuf> {
+    let scarb_metadata = MetadataCommand::new().inherit_stderr().exec();
+    let scarb_metadata = scarb_metadata?;
+    let workspace_root = scarb_metadata.workspace.root.clone();
+    let cache_dir_path = workspace_root.join(CACHE_DIR);
+    std::fs::create_dir_all(&cache_dir_path)?;
+    Ok(cache_dir_path)
+}
+
+pub fn write_failed_tests(all_failed_tests: &[TestCaseSummary]) -> Result<()> {
+    let tests_failed_path = cache_dir()?.join(".prev_tests_failed");
+    let mut file = std::fs::OpenOptions::new()
+        .write(true)
+        .create(true)
```

Comment:
Shouldn't we use `truncate` here too? 
https://stackoverflow.com/questions/65782872/how-to-delete-the-contents-of-a-file-before-writing-to-it

Reply:
Probably I can use `https://doc.rust-lang.org/std/fs/struct.File.html#method.create` instead